packages/back-end/src/validators/safe-rollout.ts

```diff
@@ -36,7 +37,15 @@ const safeRolloutNotification = [
   "rollback",
 ] as const;
 export type SafeRolloutNotification = typeof safeRolloutNotification[number];
-
+const rampUpSchedule = z.object({
+  enabled: z.boolean(),
+  step: z.number(),
+  steps: z.array(z.number()), // percentage of traffic to release each step
+  nextUpdate: z.date().optional(),
+  lastUpdate: z.date().optional(),
+  rampUpCompleted: z.boolean(),
```

Comment:
Should we keep track of when the ramp up changes happened? For example, went to step 2 at date X and then step 3 at date Y.